backend/Testing/Fixtures/IntegrationFixture.cs

```diff
@@ -45,9 +43,12 @@ private static void DeletePreviousTestFiles()
 
     private void InitTemplateRepo()
     {
-        if (TemplateRepo.Exists) return;
-        using var stream = TemplateRepoZip.OpenRead();
-        ZipFile.ExtractToDirectory(stream, TemplateRepo.FullName);
+        lock (TemplateRepo)
+        {
+            if (TemplateRepo.Exists) return;
+            using var stream = TemplateRepoZip.OpenRead();
+            ZipFile.ExtractToDirectory(stream, TemplateRepo.FullName);
```

Comment:
should refresh the status of the template repo object after we unzip so we don't extract twice and cause issues.
```suggestion
            ZipFile.ExtractToDirectory(stream, TemplateRepo.FullName);
            TemplateRepo.Refresh();
```